Otus-DevOps-2018-02/ifqthenp_infra
# Homework 10. Deployment and configuration management with Ansible

# Completed Homework № 10

:large_blue_diamond: [x] Main tasks
:large_orange_diamond: [x] Extra tasks for self-study

## The following tasks have been completed

- :large_blue_diamond: Created several ansible playbooks for Reddit application deployment
  - `reddit_app_one_play` contains all necessary tasks for deployment in one play separated with tags
  - `reddit_app_multiple_plays` has the same functionality as playbook above but divided in three plays
  - `app.yml`, `db.yml`, and `deploy.yml` are three playbooks containing tasks for specific group of hosts. The playbooks imported in `site.yml` in predefined order to deploy Reddit application.
  - `packer_db.yml` and `packer_app.yml` are the playbooks for packer images provisioning using ansible
- :large_blue_diamond: Changed provisioning config in `packer/app.json` and `packer/db.json` to replace bash scripts with ansible playbooks
- :large_orange_diamond: Created configuration for dynamic inventory using `gce.py`

## How to get project running

To deploy Reddit application `cd` into `ansible` directory and run `ansible-playbook site.yml`

## How to check that project actually works

- Ensure that the build passed TravisCI tests
  - [#andrei_bogomja](https://devops-team-otus.slack.com/messages/C9M6L1XB3) Slack channel
  - actual GitHub pull request page
- Type in the console `ansible all -m ping` to see if ping is successful
- Type in the console `gcloud compute instances list` and enter in the browser url bar `http://{{reddit-app EXTERNAL_IP}}:9292`

## PR checklisk
